### PR TITLE
feature/18: Added support for error-code-report 1.0.0 syntax

### LINKED ISSUE #18: Support error-code-report 1.0.0 syntax
Support error-code-report 1.0.0 syntax

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #18

### CODE REVIEW

File: src/main/java/com/exasol/errorcodecatalog/collector/GraphQlClient.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    private static String quote(final String nextPage) {
+        if (nextPage.contains("\"")) {
+            throw new IllegalStateException(ExaError.messageBuilder("F-EC-15")
+                    .message("String contained illegal char \".").ticketMitigation().toString());
+        }
+        return "\"" + nextPage + "\"";
+    }
+
+    /**
+     * List all public repos of the exasol integration team.
```
Comment (morazow) [inline src/main/java/com/exasol/errorcodecatalog/collector/GraphQlClient.java:56]:
```suggestion
     * List all public repos of the Exasol integration team.
```

File: src/main/java/com/exasol/errorcodecatalog/collector/GraphQlClient.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    private byte[] toJson(final JsonValue requestJson) {
+        try (final ByteArrayOutputStream buffer = new ByteArrayOutputStream()) {
+            try (final JsonWriter writer = Json.createWriter(buffer)) {
+                writer.write(requestJson);
+            }
+            return buffer.toByteArray();
+        } catch (final IOException exception) {
+            throw new UncheckedIOException(
+                    ExaError.messageBuilder("F-EC-13").message("Exception while serializing to JSON.").toString(),
```
Comment (morazow) [inline src/main/java/com/exasol/errorcodecatalog/collector/GraphQlClient.java:209]:
```suggestion
                    ExaError.messageBuilder("F-EC-13").message("Failed to serialize to JSON.").toString(),
```

File: src/main/java/com/exasol/errorcodecatalog/renderer/ProjectVersion.java [hunk 3]
```diff
@@ -2,9 +2,17 @@
...
 import java.util.Arrays;
 
+/**
+ * This class represents a version number.
+ */
 public class ProjectVersion implements Comparable<ProjectVersion> {
     private final String[] parts;
 
+    /**
+     * Create a nwq instance of {@link ProjectVersion}.
```
Comment (morazow) [inline src/main/java/com/exasol/errorcodecatalog/renderer/ProjectVersion.java:12]:
```suggestion
     * Create a new instance of {@link ProjectVersion}.
```

**Review by morazow — CHANGES_REQUESTED** [review]
Left minor suggestions

File: pom.xml [hunk 4]
```diff
@@ -245,10 +246,16 @@
                 <configuration>
                     <modules>
                         <module>jar_artifact</module>
+                        <module>lombok</module>
                     </modules>
                     <excludedFiles>
-                        <excludedFile>.travis.yml</excludedFile>
+                        <excludedFile>.github/workflows/ci-build-next-java.yml</excludedFile>
```
Comment (morazow) [inline pom.xml:252]: File seems to be exactly same file, do you need to exclude it in project-keeper?

**Review by morazow — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (jakobbraun) [reply pom.xml:252]: I added additional github token:
image

**Review by morazow — APPROVED** [review]

**Pull request merged**